app/frontend/courses/curricula/components/CoursesCurriculum__Overlay.res

```diff
@@ -910,11 +910,15 @@ let make = (
   React.useEffect1(loadTargetDetails(target, send), [Target.id(target)])
   // Load peer submissions only if the target has discussion enabled and the current user is a participant.
   React.useEffect1(() => {
-    if currentUser->CurrentUser.isParticipant {
-      reloadSubmissions(send, target->Target.id)
+    switch state.targetDetails {
+    | Some(targetDetails) =>
+      if currentUser->CurrentUser.isParticipant && targetDetails->TargetDetails.discussion {
+        reloadSubmissions(send, target->Target.id)
+      }
+    | None => ()
     }
     None
-  }, [Target.id(target)])
+  }, [state.targetDetails])
```

Comment:
I can't tell why this change was required. 🤔

Reply:
This was changed to https://github.com/pupilfirst/pupilfirst/commit/ac236c31861111d8bcb3a370bef777d70ee39ef3 by Hari after discussion.